### PR TITLE
Get AccessToken during auth flow

### PR DESCRIPTION
**Pull request by ermel272**
### Description
Added code to query twitter for user AccessToken so that user based twitter queries will be authorized.

### Changelog
- Add TwitterSession for storage of AccessToken
- Add request to Twitter to receive AccessToken from oauth_verifier

### Addresses Issues
#49

### CODE REVIEW

File: app/src/main/java/com/dankideacentral/dic/authentication/LoginActivity.java [hunk 1]
```diff
@@ -66,15 +66,9 @@ protected void onPostExecute(RequestToken requestToken) {
...
-            SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
-            SharedPreferences.Editor prefEditor = preferences.edit();
-            prefEditor.putString(getString(R.string.twitter_auth_preference), requestToken.getToken());
-            prefEditor.putString(getString(R.string.twitter_auth_secret_preference), requestToken.getTokenSecret());
-            prefEditor.commit();
 
-            // Re-initialize the twitter singleton with the stored authentication tokens
-            TwitterUtil.init(getApplicationContext());
+            // Give requestToken to the TwitterSession
+            TwitterSession.getInstance().setRequestToken(requestToken);
```
Comment (srowhani) [inline app/src/main/java/com/dankideacentral/dic/authentication/LoginActivity.java:71]: TwitterSession isn't imported in this file. Does this work?

File: app/src/main/java/com/dankideacentral/dic/authentication/OAuthWebViewFragment.java [hunk 2]
```diff
@@ -28,10 +27,16 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
             @Override
             public boolean shouldOverrideUrlLoading(WebView view, String url) {
                 if (url.contains("oauth_verifier=")) {
-                    Intent intent = new Intent(getActivity().getApplicationContext(), TweetFeedActivity.class);
-                    intent.setData(Uri.parse(url));
+                    // Create user session
+                    TwitterSession.getInstance().createSession(Uri.parse(url),
```
Comment (srowhani) [inline app/src/main/java/com/dankideacentral/dic/authentication/OAuthWebViewFragment.java:39]: Again, shouldn't we have to import this?

**Review by srowhani — APPROVED** [review]
Looks good, I like the application of singleton pattern.

**Comment by bazadactyl** [comment]
Looks good to me.

File: app/src/main/java/com/dankideacentral/dic/authentication/LoginActivity.java (hunk 1 above)
Comment (ermel272) [reply app/src/main/java/com/dankideacentral/dic/authentication/LoginActivity.java:71]: Doesn't need to be due to the Singleton implementation

File: app/src/main/java/com/dankideacentral/dic/authentication/OAuthWebViewFragment.java (hunk 2 above)
Comment (ermel272) [reply app/src/main/java/com/dankideacentral/dic/authentication/OAuthWebViewFragment.java:39]: getInstance() is a static method so inline calls don't require an import. Sorry - just realized my previous comment didn't clear this up at all

**Pull request merged**